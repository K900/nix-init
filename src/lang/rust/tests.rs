use heck::AsKebabCase;
use insta::assert_debug_snapshot;
use tempfile::tempdir;

use super::resolve_workspace;
use crate::{inputs::AllInputs, lang::rust::load_rust_dependencies};

use std::fs::{copy, create_dir, File};

macro_rules! tests {
    ($($name:ident)+) => {
        $(
            #[test]
            fn $name() {
                assert_debug_snapshot!(run(stringify!($name)));
            }
        )+
    };
}

tests! {
    libz_ng
    libz_static
    libz_stock
    llvm
    zstd_env
    zstd_old
    zstd_pkg_config
}

fn run(name: &str) -> AllInputs {
    let dir = tempdir().unwrap();
    let dir = dir.path();

    let fixture = format!("src/lang/rust/fixtures/{}", AsKebabCase(name));
    copy(format!("{fixture}.toml"), dir.join("Cargo.toml")).unwrap();
    copy(format!("{fixture}-lock.toml"), dir.join("Cargo.lock")).unwrap();

    let src = dir.join("src");
    create_dir(&src).unwrap();
    File::create(src.join("lib.rs")).unwrap();

    let mut inputs = AllInputs::default();
    load_rust_dependencies(&mut inputs, &resolve_workspace(dir).unwrap());
    inputs
}
