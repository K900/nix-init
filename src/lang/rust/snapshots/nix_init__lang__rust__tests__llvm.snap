---
source: src/lang/rust/tests.rs
expression: run(stringify!(llvm))
---
AllInputs {
    native_build_inputs: Inputs {
        always: {},
        darwin: {},
        aarch64_darwin: {},
        x86_64_darwin: {},
        linux: {},
        aarch64_linux: {},
        x86_64_linux: {},
    },
    build_inputs: Inputs {
        always: {},
        darwin: {},
        aarch64_darwin: {},
        x86_64_darwin: {},
        linux: {},
        aarch64_linux: {},
        x86_64_linux: {},
    },
    env: {
        "LLVM_SYS_150_PREFIX": (
            "llvmPackages_15.llvm.dev",
            [
                "llvmPackages_15",
            ],
        ),
    },
}
